src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReferenceEqualsEqualityComparer.cs

```diff
@@ -21,7 +22,7 @@ bool IEqualityComparer<T>.Equals(T x, T y)
 
         int IEqualityComparer<T>.GetHashCode(T obj)
         {
-            return obj!.GetHashCode();
+            return RuntimeHelpers.GetHashCode(obj!);
```

Comment:
I don't think the bang is necessary. `IEqualityComparer<T>` disallows nulls:
https://source.dot.net/#System.Private.CoreLib/IEqualityComparer.cs,15

Reply:
It looks like is necessary, you need to apply null forgiving (`!`) since `obj` can be null, which is forbidden on `RuntimeHelpers.GetHashCode(object obj)` . That is also why the CI failed.

Reply:
Gotcha. Good catch @Jozkee, thanks.